src/screens/twinClassFields/twin-class-fields.tsx

```diff
@@ -1,103 +1,162 @@
+import { z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
 
 const colDefs: Record<
   keyof Pick<
-    TwinClassField,
-    "id" | "key" | "name" | "description" | "required"
+    TwinClassFieldV2_DETAILED,
+    | "id"
+    | "twinClassId"
+    | "key"
+    | "name"
+    | "description"
+    | "fieldTyperFeaturerId"
+    | "viewPermissionId"
+    | "editPermissionId"
+    | "required"
   >,
-  ColumnDef<TwinClassField>
+  ColumnDef<TwinClassFieldV2_DETAILED>
 > = {
   id: {
     accessorKey: "id",
     header: "ID",
     cell: (data) => <GuidWithCopy value={data.getValue<string>()} />,
   },
+
+  twinClassId: {
+    id: "twinClassId",
+    accessorKey: "twinClassId",
+    header: "Class",
+    cell: ({ row: { original } }) =>
+      original.twinClass && (
+        <div className="max-w-48 inline-flex">
+          <TwinClassResourceLink data={original.twinClass} withTooltip />
+        </div>
+      ),
+  },
+
   key: {
     accessorKey: "key",
     header: "Key",
   },
+
   name: {
     accessorKey: "name",
     header: "Name",
   },
+
   description: {
     accessorKey: "description",
     header: "Description",
   },
+
+  fieldTyperFeaturerId: {
+    accessorKey: "fieldTyperFeaturerId",
+    header: "Field typer",
+    cell: ({ row: { original } }) =>
+      original.fieldTyperFeaturer && (
+        <div className="max-w-48 inline-flex">
+          <FeaturerResourceLink
+            data={original.fieldTyperFeaturer}
+            withTooltip
+          />
+        </div>
+      ),
+  },
+
+  viewPermissionId: {
+    accessorKey: "viewPermissionId",
+    header: "View permission",
+    cell: ({ row: { original } }) =>
+      original.viewPermission && (
+        <div className="max-w-48 column-flex space-y-2">
+          <PermissionResourceLink data={original.viewPermission} withTooltip />
+        </div>
+      ),
+  },
+
+  editPermissionId: {
+    accessorKey: "editPermissionId",
+    header: "Edit permission",
+    cell: ({ row: { original } }) =>
+      original.editPermission && (
+        <div className="max-w-48 column-flex space-y-2">
+          <PermissionResourceLink data={original.editPermission} withTooltip />
+        </div>
+      ),
+  },
+
   required: {
     accessorKey: "required",
     header: "Required",
     cell: (data) => <>{data.getValue() && <Check />}</>,
   },
 };
 
-const twinClassFieldchema = z.object({
-  key: z
-    .string()
-    .min(1)
-    .max(100)
-    .regex(
-      REGEX_PATTERNS.ALPHANUMERIC_WITH_DASHES,
-      "Key can only contain latin letters, numbers, underscores and dashes"
-    ),
-  name: z.string().min(0).max(100),
-  description: z.string(),
-  required: z.boolean(),
-  fieldTyperFeaturerId: z.number(),
-  fieldTyperParams: z.record(z.string()),
-});
-
-export function TwinClassFields() {
-  const api = useContext(ApiContext);
-  const { twinClass } = useContext(TwinClassContext);
+export function TwinClassFields({ twinClassId }: { twinClassId?: string }) {
```

Comment:
please remove `({ twinClassId }: { twinClassId?: string }`

we don't pass `twinClassId` here